pkg/util/tls/tls.go

```diff
@@ -0,0 +1,34 @@
+package tls
```

Comment:
I think we are copying this file and the tests from master, but this version is truncated... do we want to copy the whole thing over? I think as written the tests won't pass because e.g. `NewCA` isn't defined here but there's a corresponding test.